src/libraries/System.Threading.RateLimiting/tests/Infrastructure/Utils.cs

```diff
@@ -30,10 +30,16 @@ internal static class Utils
         // Gets and runs the Heartbeat function on the DefaultPartitionedRateLimiter
         internal static Task RunTimerFunc<T>(PartitionedRateLimiter<T> limiter)
         {
-            var innerTimer = limiter.GetType().GetField("_timer", BindingFlags.NonPublic | BindingFlags.Instance);
+            var limiterType = limiter.GetType();
+
+            // Trick trimming into seeing what type we're operating on
+            if (string.Empty.Length > 0)
+                limiterType = Type.GetType("System.Threading.RateLimiting.DefaultPartitionedRateLimiter`2, System.Threading.RateLimiting");
```

Comment:
```suggestion
            var expectedLimiterType = Type.GetType("System.Threading.RateLimiting.DefaultPartitionedRateLimiter`2, System.Threading.RateLimiting");
            Assert.Equal(expectedLimiterType, limiterType);
```
We can validate the assumptions as part of the test.

Reply:
That would still not fix the dataflow analysis because the analysis doesn't understand limiterType is now same as expectedLimiterType. I'm becoming allergic to these kinds of tests because it has been the same issues for 3 weeks now.

Reply:
I understand that it would help dataflow analysis, but it would eagerly catch cases where somebody renames the type that is used as internal implementation detail.